client/src/test/index.ts

```diff
@@ -0,0 +1,39 @@
+import path from 'path';
+import Mocha from 'mocha';
+import glob from 'glob';
+
+export function run(): Promise<void> {
+	// Create the mocha test
+	const mocha = new Mocha({
+		ui: 'tdd',
+	});
+	mocha.useColors(true);
+	mocha.timeout(100000);
+
+	const testsRoot = __dirname;
+
+	return new Promise((resolve, reject) => {
+		glob('**.test.js', { cwd: testsRoot }, (err, files) => {
+			if (err) {
+				return reject(err);
+			}
+
+			// Add files to the test suite
+			files.forEach(f => mocha.addFile(path.resolve(testsRoot, f)));
+
+			try {
+				// Run the mocha test
+				mocha.run(failures => {
+					if (failures > 0) {
+						reject(new Error(`${failures} tests failed.`));
+					} else {
+						resolve();
+					}
+				});
+			} catch (err) {
+				console.error(err);
+				reject(err);
+			}
```

Comment:
Is this file necessary? Seems like something you could stick in your `package.json`

```json
{
  "scripts": {
     "test": "mocha ./**/*.test.js"
  }
}
```

Or are you looking for a way to just do `./node_modules/.bin/mocha dist/test/index.js`?

Reply:
I'm not sure what you mean by this: ` ./node_modules/.bin/mocha dist/test/index.js`. I tried to run the tests by running the test script as you suggested, but that gave a ReferenceError, saying 'jest' is not defined. 
I think the file is necessary to run all tests, and show if errors occurred, but I'm not not 100% sure

Reply:
Now I'm confused. This script is setting up Mocha, Jest seems unrelated.

Reply:
so I do think I need the ./index, because in runTests, a path has to be specified for `vscode-test` for `extensionTestPath`

Reply:
in the docs I found this: 
`Absolute path to the extension tests runner. Passed to --extensionTestsPath. Can be either a file path or a directory path that contains an index.js. Must export a run function of the following signature:

function run(): Promise<void>;
When running the extension test, the Extension Development Host will call this function that runs the test suite. This function should throws an error if any test fails.`

Reply:
Ahhh okay, so VSCode itself is running this file. Okay sounds good. Thanks for looking into this!

Reply:
I'm still a bit confused about the Jest part, unless Jest has some Mocha adapters 🤔